azure/scope/cluster.go

```diff
@@ -1006,3 +1008,58 @@ func (s *ClusterScope) TagsSpecs() []azure.TagsSpec {
 		},
 	}
 }
+
+// PrivateEndpointSpecs returns the private endpoint specs.
+func (s *ClusterScope) PrivateEndpointSpecs() []azure.ResourceSpecGetter {
+	numberOfSubnets := len(s.AzureCluster.Spec.NetworkSpec.Subnets)
+	if s.IsAzureBastionEnabled() {
+		numberOfSubnets++
+	}
+
+	privateEndpointSpecs := make([]azure.ResourceSpecGetter, 0, numberOfSubnets)
```

Comment:
There can be more than 1 privateEndpoint per subnet so can't this slice end up being larger than the numberOfSubnets?

Reply:
That is just the initial capacity of the slice. `append` will expand it if the len of the slice is getting bigger than the capacity

Reply:
ah, my bad. I thought it was a fixed size when it was done this way. Just my inexperience with Go.